src/accountsdb/swiss_map.zig

```diff
@@ -273,7 +277,22 @@ pub fn SwissMapUnmanaged(
             value_ptr: *Value,
         };
 
-        pub fn remove(self: *@This(), key: Key) error{KeyNotFound}!void {
+        pub fn remove(self: *Self, key: Key) error{KeyNotFound}!void {
+            return @call(.always_inline, removeImpl, .{ self, key, .void });
+        }
+
+        pub fn fetchRemove(self: *Self, key: Key) error{KeyNotFound}!Value {
+            return @call(.always_inline, removeImpl, .{ self, key, .value });
+        }
+
+        fn removeImpl(
+            self: *@This(),
+            key: Key,
+            comptime ret: enum { void, value },
+        ) error{KeyNotFound}!switch (ret) {
+            .void => void,
+            .value => Value,
+        } {
```

Comment:
the less comptime zig specific things we can do the better - should just have fetchRemove named as remove, and we can ignore the value if we dont care about it

Reply:
9a7da9775ad6d8213c3d81aad892d5e7def41ef3